lib/num2str.c

```diff
@@ -112,6 +112,9 @@ char *num2str(uint64_t num, int maxlen, int base, int pow2, enum n2s_unit units)
 	sprintf(tmp, "%.*f", (int)(maxlen - strlen(tmp) - 1),
 		(double)modulo / (double)thousand);
 
+	if (tmp[0] == '1')
+		num++;
```

Comment:
OK so if the modulo / thousand has been rounded up to 1 when printing it as a float then we assume we have to do a carry. Could you add a test to `unittests/lib/num2str.c` to ensure this behaviour is preserved?